tests/src/Network/CurlResultTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Friendica\Test\src\Network;
+
+use Friendica\Network\CurlResult;
+use PHPUnit\Framework\TestCase;
+
+class CurlResultTest extends TestCase
+{
+	public function setUp()
+	{
+		parent::setUp();
+
+		require_once __DIR__.'/../../../boot.php';
+		require_once __DIR__.'/../../../include/text.php';
+	}
+
+	/**
+	 * @small
+	 */
+	public function testNormal()
+	{
+		$header = file_get_contents(__DIR__ . '/../../datasets/curl/about.head');
+		$body = file_get_contents(__DIR__ . '/../../datasets/curl/about.body');
+
+
+		$curlResult = new CurlResult('https://test.local', $header . $body, [
+			'http_code' => 200,
+			'content_type' => 'text/html; charset=utf-8',
+			'url' => 'https://test.local'
+		]);
+
+		$this->assertTrue($curlResult->isSuccess());
+		$this->assertFalse($curlResult->isTimeout());
+		$this->assertFalse($curlResult->isRedirectUrl());
+		$this->assertSame($header, $curlResult->getHeader());
+		$this->assertSame($body, $curlResult->getBody());
+		$this->assertSame('text/html; charset=utf-8', $curlResult->getContentType());
+		$this->assertSame('https://test.local', $curlResult->getUrl());
+		$this->assertSame('https://test.local', $curlResult->getRedirectUrl());
+	}
+
+	/**
+	 * @small
+	 */
+	public function testRedirect()
+	{
+		$header = file_get_contents(__DIR__ . '/../../datasets/curl/about.head');
+		$body = file_get_contents(__DIR__ . '/../../datasets/curl/about.body');
+
+
+		$curlResult = new CurlResult('https://test.local/test/it', $header . $body, [
+			'http_code' => 301,
+			'content_type' => 'text/html; charset=utf-8',
+			'url' => 'https://test.local/test/it',
+			'redirect_url' => 'https://test.other'
+		]);
+
+		$this->assertTrue($curlResult->isSuccess());
+		$this->assertFalse($curlResult->isTimeout());
+		$this->assertTrue($curlResult->isRedirectUrl());
+		$this->assertSame($header, $curlResult->getHeader());
+		$this->assertSame($body, $curlResult->getBody());
+		$this->assertSame('text/html; charset=utf-8', $curlResult->getContentType());
+		$this->assertSame('https://test.local/test/it', $curlResult->getUrl());
+		$this->assertSame('https://test.other/test/it', $curlResult->getRedirectUrl());
```

Comment:
How is the final redirect URL different from the one you fed it? Why is the old URL path arbitrarily slapped on top of it?

Reply:
it redirects from `https://test.**local**` to `https://test.**other**`

the path is because of 
https://github.com/friendica/friendica/blob/fc19373ec4d167933255d48fa58771127f8b4c9a/src/Network/CurlResult.php#L162-L164

Reply:
But wouldn't the redirect URL be a fully-qualified one including path? Why do you assume that if there's no path it should be the previous URL path?

Reply:
To be honest, it was in the original code which I moved to the check.

I thought it's a fallback for a miss-configured server. It doesn't work for Redirects in the header too, so it's just half the way ;-)

Reply:
Fair enough.